esp-hal/src/soc/esp32s3/psram.rs

```diff
@@ -179,7 +179,10 @@ pub(crate) fn init_psram(config: PsramConfig) {
         // https://docs.espressif.com/projects/esp-idf/en/stable/esp32s3/api-reference/system/mm.html#introduction
         let mmu_table_ptr = DR_REG_MMU_TABLE as *const u32;
         let mut mapped_pages = 0;
-        for i in (0..FLASH_MMU_TABLE_SIZE).rev() {
+
+        // the bootloader is using the last page to access flash internally
+        // (e.g. to read the app descriptor) so we just skip that
+        for i in (0..(FLASH_MMU_TABLE_SIZE - 1)).rev() {
```

Comment:
This logic is hard coded for the esp-idf bootloader, should it go behind a feature/config?

Reply:
In theory probably yes - but I'm not sure it will have any real-world implications 🤔

Reply:
Ideally this code would be in the bootloader support crate anyway, so I don't think we should further complicate this for now.